course_preference.py

```diff
@@ -34,29 +33,30 @@ def save_preference():
 
     preferences = data['preferences']
```

Comment:
```suggestion
    preferences = data.get('preferences', None)
```

You should also consider the case the key is not defined.